### PR TITLE
fix(datagrid): implement DefaultDateCellRenderer and use int renderer for all number type

### PR DESCRIPTION
**Pull request by jare-talend**
**What is the problem this PR is trying to solve?**
we have to show a date in ISO 8601 format when the schema has a "logicalType" property to 'timestamp-millis'
For all number or float type (int, long, double, float), we have to show the int to left to right

**What is the chosen solution to this problem?**
implement a DefaultDateCellRenderer to manage this type.
implement a mapping for the number or float type to choose the int renderer

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/datagrid/src/components/DefaultCellRenderer/DefaultValueRenderer.component.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ export default class DefaultValueRenderer extends React.Component {
 	render() {
 		let stringValue;
 
-		if (this.props.value === null) {
+		if (this.props.value === null || this.props.value === undefined) {
```
Comment (romainseb) [inline packages/datagrid/src/components/DefaultCellRenderer/DefaultValueRenderer.component.js]: Is this should be enought ?
```suggestion
		if (this.props.value == null) {
```

File: packages/datagrid/src/components/DefaultDateCellRenderer/DefaultDateCellRenderer.component.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import DefaultValueRenderer from '../DefaultCellRenderer/DefaultValueRenderer.component';
+
+export default function DefaultDateCellRenderer({ data }) {
```
[the remark this answers is not in the document]
Comment (jare-talend) [reply packages/datagrid/src/components/DefaultDateCellRenderer/DefaultDateCellRenderer.component.js]: The cell value can be invalid or null. It is a case expected. So, it isn't relevant to trigger a Error in the case. So, new Date can throw an error if the type is invalid, I catch it to set the original value.

File: packages/datagrid/src/components/DefaultCellRenderer/mappingTypeRenderer.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { PRIMITIVES_MAPPING, LOGICAL_MAPPING } from '../../constants/avro-type.constant';
+
+// eslint-disable-next-line import/prefer-default-export
```
Comment (romainseb) [inline packages/datagrid/src/components/DefaultCellRenderer/mappingTypeRenderer.js]: ( i dislike this rule 😏  )

File: packages/datagrid/src/constants/avro-type.constant.js [hunk 4]
```diff
@@ -1 +1,12 @@
-export default ['boolean', 'int', 'long', 'float', 'double', 'bytes', 'string'];
+export const AVRO_TYPES = ['boolean', 'int', 'long', 'float', 'double', 'bytes', 'string'];
+export const LOGICAL_TYPES = ['timestamp-millis'];
```
Comment (romainseb) [inline packages/datagrid/src/constants/avro-type.constant.js]: Is the only logical type for the dates ?

File: packages/datagrid/src/components/DefaultCellRenderer/DefaultValueRenderer.component.js (hunk 1 above)
Comment (jare-talend) [reply packages/datagrid/src/components/DefaultCellRenderer/DefaultValueRenderer.component.js]: exact, with abstract equality undefined == null && null == null. But this rule is often bad understand. So, i checked this to be sure https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#Loose_equality_using ;)

File: packages/datagrid/src/constants/avro-type.constant.js (hunk 4 above)
Comment (jare-talend) [reply packages/datagrid/src/constants/avro-type.constant.js]: yes

File: packages/datagrid/src/components/DefaultDateCellRenderer/DefaultDateCellRenderer.component.test.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import DefaultDateCellRenderer from './DefaultDateCellRenderer.component';
+import DefaultValueRenderer from '../DefaultCellRenderer/DefaultValueRenderer.component';
+
+describe('#DefaultDateCellRenderer', () => {
+	it('should render DefaultDateCellRenderer with DefaultValueRenderer', () => {
+		const wrapper = shallow(<DefaultDateCellRenderer data={{ value: 1511873062123 }} />);
```
[the remark this answers is not in the document]
Comment (jare-talend) [reply packages/datagrid/src/components/DefaultDateCellRenderer/DefaultDateCellRenderer.component.test.js]: There is already a test to test a string value and check that this value isn't modify.

File: packages/datagrid/src/components/DefaultCellRenderer/DefaultValueRenderer.component.js (hunk 1 above)
Comment (romainseb) [reply packages/datagrid/src/components/DefaultCellRenderer/DefaultValueRenderer.component.js]: Alright, i don't care that much on this point ;)

File: packages/datagrid/src/components/DefaultDateCellRenderer/DefaultDateCellRenderer.component.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (romainseb) [reply packages/datagrid/src/components/DefaultDateCellRenderer/DefaultDateCellRenderer.component.js]: As we have discussed it by voice, we should not keep the invariant as it's a business case that the data we have could throw this exception.
We don't want to flood the console with some wrong data to display in the datagrid ( that could be a classic use case of the datagrid :P )

File: packages/datagrid/src/components/DefaultDateCellRenderer/DefaultDateCellRenderer.component.test.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (nicomaligne) [reply packages/datagrid/src/components/DefaultDateCellRenderer/DefaultDateCellRenderer.component.test.js]: I meant that certain string format are interpreted : https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Date/toISOString

[the remark this answers is not in the document]
Comment (nicomaligne) [reply packages/datagrid/src/components/DefaultDateCellRenderer/DefaultDateCellRenderer.component.test.js]: Ok avro spec need a int or long sorry.

**Pull request merged**